tools/repl/src/repl.rs

```diff
@@ -40,6 +42,24 @@ macro_rules! next_argument {
     };
 }
 
+macro_rules! next_argument_or_result {
+    ($arg_name:ident, $args:ident, $error_msg:expr) => {
+        let $arg_name = if let Some($arg_name) = $args.next() {
+            $arg_name
+        } else {
+            return Err(String::from($error_msg));
+        };
+    };
+}
+
+struct CallModuleArguments<'a> {
```

Comment:
Consider use more meaningful name to the lifetime like `'args` (or smth like that), because this structs will be used only for parse from raw arguments.